feat(cactus-i18n): restrict logging to debugMode only

JamesNimlos opened the pull request:
* removes all logging when not in debug mode
* allows extra properties to be passed via `I18nSection` to the controller.load function
* allows adding dependencies to an `I18nSection`

[CACTUS-232]
[CACTUS-233]

[CACTUS-232]: https://repayonline.atlassian.net/browse/CACTUS-232
[CACTUS-233]: https://repayonline.atlassian.net/browse/CACTUS-233

Dhalton commented on modules/cactus-i18n/src/BaseI18nController.ts:
```diff
@@ -234,6 +240,24 @@ export default abstract class BaseI18nController {
       })
   }
 
+  hasText({
```
Was a hasText function something that was asked for by the Channels guys? Just curious what their use-case is

JamesNimlos replied:
Yeah, we're going to use it to decide whether to render the I18nText component or pull text from a config

Dhalton replied:
Ah, right okay that makes sense. Thanks!

Dhalton commented:
What do you think about adding an extra test where you pass a dependency as an object with its own "props"?

JamesNimlos commented:
@Dhalton I've added the test you mentioned and I've also split the tests to run in different steps in the CI so hopefully that mitigates some issues we've seen with inconsistent test runs.
